### PR TITLE
editorial: Clarify DependencyLevels, SlsaResult, and quote level keys.

### PR DESCRIPTION
**Pull request by AdamZWu**
This is related to issues #968, just a cleanup preparing the proposal to remove "FAILED" at v1.1 in a future commit.

### CODE REVIEW

**Comment by AdamZWu** [comment]
Oh, hold on. `SLSA_BUILD_LEVEL_UNEVALUATED` got leaked from v1.1 into v1.0.
Let me fix that.

File: docs/spec/v1.0/verification_summary.md [hunk 1]
```diff
@@ -191,12 +191,15 @@ of the other top-level fields, such as `subject`, see [Statement]._
 >
 > Map from [SlsaResult] to the number of the artifact's _transitive_ dependencies
 > that were verified at the indicated level. Absence of a given level of
-> [SlsaResult] MUST be interpreted as reporting _0_ dependencies at that level.
+> [SlsaResult] MUST be interpreted as reporting _0_ dependencies at that level,
+> e.g. an empty `dependencyLevels` object means that the artifact has **no**
+> dependency at all (vs. a non-present, i.e. "null", `dependencyLevels` means
+> that the verifier does not disclose the dependency stats).
```
Comment (MarkLodato) [inline docs/spec/v1.0/verification_summary.md:197]:
```suggestion
> [SlsaResult] MUST be interpreted as reporting _0_ dependencies at that level.
> An set-but-empty `dependencyLevels` object means that the artifact has **no**
> dependency at all, while an unset or null `dependencyLevels` means that the
> verifier makes no claims about the artifact's dependencies.
```

Minor wording suggestion.

File: docs/spec/v1.0/verification_summary.md [hunk 2]
```diff
@@ -191,12 +191,15 @@ of the other top-level fields, such as `subject`, see [Statement]._
...
 >
 > Users MUST count each dependency only once per SLSA track, at the highest
-> level verified. For example, if a dependency meets SLSA_BUILD_LEVEL_2,
-> you include it with the count for SLSA_BUILD_LEVEL_2 but not the count for
->SLSA_BUILD_LEVEL_1.
+> level verified. For example, if a dependency meets `SLSA_BUILD_LEVEL_2`,
+> you include it with the count for `SLSA_BUILD_LEVEL_2` but not the count for
+> the other `SLSA_BUILD` levels.
```
Comment (MarkLodato) [inline docs/spec/v1.0/verification_summary.md:202]: I find this change to be less readable. What was the motivation? L1 is the only level below L2, so it seemed clear to me that this was explaining that if it's L2, you don't also report it at L1.

File: docs/spec/v1.0/verification_summary.md [hunk 3]
```diff
@@ -252,17 +255,18 @@ WARNING: This is just for demonstration purposes.
...
--   SLSA_BUILD_LEVEL_0
--   SLSA_BUILD_LEVEL_1
--   SLSA_BUILD_LEVEL_2
--   SLSA_BUILD_LEVEL_3
--   FAILED (Indicates policy evaluation failed)
-
-Note that each SLSA level implies the levels below it. For example,
-SLSA_BUILD_LEVEL_3 means (SLSA_BUILD_LEVEL_1 + SLSA_BUILD_LEVEL_2 +
-SLSA_BUILD_LEVEL_3).
+SHOULD contain one value per SLSA track:
```
Comment (MarkLodato) [inline docs/spec/v1.0/verification_summary.md:258]: This looks like a mistaken content change. This PR is just supposed to change formatting and minor wording changes. However, this change affects the schema, but also doesn't work. The field is a scalar String, but the new wording suggest it's a collection of strings.

**Review by MarkLodato — CHANGES_REQUESTED** [review]
Thanks!

File: docs/spec/v1.0/verification_summary.md (hunk 2 above)
Comment (AdamZWu) [reply docs/spec/v1.0/verification_summary.md:202]: Oh good point, I think I overlooked that this is already in the context of "for example". :P

File: docs/spec/v1.0/verification_summary.md (hunk 3 above)
Comment (AdamZWu) [reply docs/spec/v1.0/verification_summary.md:258]: Oh right I think I got confused. WDYT I revise the description to clarify that the `SlsaResult` is a scalar that comprises values from multiple tracks (currently there are two: the "SLSA BUILD" track and "policy evaluation failure" track) This is to lead to the upcoming removal of the "policy evaluation failure" track values in v1.1.

Comment (MarkLodato) [reply docs/spec/v1.0/verification_summary.md]: I think it's still worse than before. Frankly, this particular field needs to be redesigned in the next version. I don't think we should bother trying to fix the v1.0 wording. Instead, we should put effort towards figuring out what a better API is. So I would revert this part of the change (in v1.0 and v1.1) and leave that to a separate PR.

Comment (AdamZWu) [reply docs/spec/v1.0/verification_summary.md]: Sounds good.

**Review by lehors — APPROVED** [review]
LGTM

**Review by trishankatdatadog — APPROVED** [review]
LGTM, thx!

**Pull request merged**